internal/driver/driver.go

```diff
@@ -198,9 +198,6 @@ func (s *Driver) getGPIO(line string) (bool, error) {
 				// waiting for gpio device fd
 				time.Sleep(1 * time.Second)
 			}
-			if err = s.setDirectionBySysfs(valid_line, "in"); err != nil {
-				return false, err
-			}
```

Comment:
I think this will work fine as long as the direction is "in" by default, but not otherwise. There could be some hardware which has the direction as "out" by default. In this case, this driver will never set it to "in", even if it the profile has `readWrite: "W"` property. I suggest adding a profile attribute `direction: out|in`, processed during the application startup, along with necessary export calls. The getGPIO and setGPIO can then just do read and write (please see also https://github.com/edgexfoundry/device-gpio/issues/19#issuecomment-1095162205)

Reply:
Did some extensive reading or GPIO with sysfs today.  I can see why sysfs is being deprecated in favor or chardev.

First - some background as I understand it/read it.  The operational setup of sysfs and GPIO is to do the following
1. export the pin
2. set the direction of the pin (in or out - with in being the default at least with Rasperry Pi)
3. perform setting/reading of the pin value as needed (as often as needed).  But if you do a write with IN direction it causes a permission denied error.  If you do a read with direction of OUT, it still works fine.
4. unexport the pin when you are done.

Step #2 should really only be done once based on what you need.  If you set the direction more than once, it resets the value of the pin to Low (or 0).

So, in my PR and per Farshid's suggestion, I am allowing for an optional additional attribute called defaultDirection (set to `out` when not provided).  This attribute is used to set the direction when the line is open.  Note: I did not do this as part of initialization because this would have required a much larger refactor and I wanted to keep the original structure/logic in place as chardev also operates this way.  In the future, someone may want to rewrite/simplify this.

Unfortunately, this still leaves some inconsistency.  In the device profile, we have a readWrite property.  Presumably, we want the readWrite property to be consistent with the GPIO pin direction (IN = Read, Out = Write).  If the defaultDirection attribute were set to `in` with the read/write set to `W` or `RW` (by mistake or because someone didn't understand GPIO) then a set gpio value operation will cause a `permission denied` error.  While this is correct in function of GPIO, it will likely lead to confusion in the context of read/write (thus I documented this in the Readme).  

Since we have a read/write property on the device profile, in the future we should probably use this to drive both the behavior and default setup of the GPIO direction to keep things consistent.  However, at the moment - the read/write property is not accessible in the driver.

I recommend that in the future, if the read/write property is set to "R", then we set the direction to `in`.  Then, their should not be a device command to set and any attempt to set the value (if it were somehow attempted in some other way) would still fail with permission denied (as it should).  If the read/write property is set to either "W" or "RW" then we set the direction to `out`.  Then both commands are available and both reads/writes will be allowed.  Note:  this is somewhat problematic in that if someone has the readWrite just set to "W", then reads should not be allowed, but I don't know that GPIO has a way to allow setting but not reading the value with sysfs.

The caveat to all this is "I think this will work" as I only have a Raspberry Pi and its implementation of GPIO to test with.  I am not sure if the same behavior is going to work with regard to setting direction to `out` and still being able to do a read in all circumstances (i.e with all sysfs implementations and all types of GPIO devices).  But the implementation seems to work for all tests I did with Rasp Pi and GPIO.

Reply:
If the direction setting is moved into initialization, I think the `readWrite` property would be available (assume this would work by querying for all the devices and getting their profile information)

I suppose another approach would be to maintain a per-pin extent of flags indicating whether we had done direction-setting yet on that pin, and only call the operation once (with this we could revert to the previous approach of, if we have a `get` first the direction should be `in`, otherwise `out` ?)

Reply:
Thanks @farshidtz, looks like I misread the PR. That said, the behavior I described will happen on the first write.

Also the README says: 

> When using sysfs, setting the direction causes the value to be reset.

 This is only true when setting direction to `out`.

The other thing I forgot to mention is that with this change, I think we've effectively removed the ability to change the direction of a pin at runtime without a corresponding change to the device profile.